include/net/lwm2m.h

```diff
@@ -200,6 +199,24 @@ typedef int (*lwm2m_engine_set_data_cb_t)(uint16_t obj_inst_id,
  */
 typedef int (*lwm2m_engine_user_cb_t)(uint16_t obj_inst_id);
 
+/**
+ * @brief Asynchronous execute notification callback.
+ *
+ * Resource executes trigger a callback of this type.
+ *
+ * Register a function of this type via:
+ * lwm2m_engine_register_exec_callback()
+ *
+ * @param[in] obj_inst_id Object instance ID generating the callback.
+ * @param[in] payload Pointer to data argument payload.
+ * @param[in] payload_len Length of argument payload in bytes.
+ *
+ * @return Callback returns a negative error code (errno.h) indicating
+ *         reason of failure or 0 for success.
+ */
+typedef int (*lwm2m_engine_execute_cb_t)(uint16_t obj_inst_id,
+                                         uint8_t *payload, uint16_t payload_len);
```

Comment:
I wonder about the naming, given the message payload is only supposed to carry arguments in execute messages, perhaps it would be more verbose to call these parameters `args` and `args_len`.